### DISCUSSION TITLE
Bézier curves with controllable t start and end values

### DISCUSSION THREAD

**Discussion by Mearleo** [cetz-package/cetz#652] [Ideas]
Can the _bezier_ function have two additional attributes for the _start_ and _end_ for the _t_ variable?

Here is the normal curve:
<img width="480" alt="image" src="https://github.com/user-attachments/assets/269dd0bf-4c6e-4b85-97d1-ad9b27bdd8dc">

And there the curve with the end of t = 0.8:
<img width="479" alt="image" src="https://github.com/user-attachments/assets/54e9a696-345d-46ef-917c-0f46448032f2">

**Comment by marisbaier** [comment]
Can this also be done with f.e. circle?

**Reply by fenjalien** [reply]
Whats an "f.e. circle"?

**Reply by marisbaier** [reply]
"for example"

**Reply by fenjalien** [reply]
*oh* probably not given the arc function exists and you don't have a start and end point for a circle

**Comment by fenjalien** [comment]
Yes-ish. If you want this as an actual feature please make an issue for it :)

I've got two ideas of how you could do it implement it:
1. The *sensible* way is to use the `shorten-path` function in `path-util`: https://github.com/cetz-package/cetz/blob/master/src/path-util.typ. You would have to update the `bezier` function and this would probably be the accepted implementation.
2. The *not*-sensible way is you can shorten it using marks. Set the stroke to be `none` and use the `pos` key to move it along the curve.

**Reply by Mearleo** [reply]
Ok, I like the second solution. But I will make an issue for the added attributes. I am new and don’t want to take my time to implement that for now…

**Reply by Mearleo** [reply]
I cannot set the stroke to none: `stroke: none` but setting it to `rgb(0,0,0,*any value*)` does the job 👍

**Reply by fenjalien** [reply]
Ah yes we don't resolve a `none` stroke properly, you can also instead set the stroke width to `0pt`
